Description: Make test runner look in bindir setting for mysql_client_test 
 To avoid polluting system bin directory, the internal test client binaries
 have been moved to mysql-testsuite's home directory. It has a setting
 to pick where to find its binaries, but doesn't correctly check this
 location for the mysql_client_test binary
Author: Lars Tangvald <lars.tangvald@oracle.com>
Bug: https://bugs.mysql.com/bug.php?id=91854
Last-Update: 2018-08-01

--- mysql-5.7-5.7.23.orig/mysql-test/mysql-test-run.pl
+++ mysql-5.7-5.7.23/mysql-test/mysql-test-run.pl
@@ -2358,6 +2358,7 @@ sub mysql_client_test_arguments(){
   my $exe;
   # mysql_client_test executable may _not_ exist
   $exe= mtr_exe_maybe_exists(vs_config_dirs('testclients', 'mysql_client_test'),
+                            "$path_client_bindir/mysql_client_test",
 			     "$basedir/testclients/mysql_client_test",
 			     "$basedir/bin/mysql_client_test");
   return "" unless $exe;
